### PR TITLE
Optimize QosManager: increase concurrency.

### PR DESCRIPTION
**Pull request by ansao-aci**
Modb notifications  are processed concurrently. Concurrent tasks are
1. updating cached information
2. programme ovsdb/flows to configure qos

### CODE REVIEW

**Review by tomflynn — CHANGES_REQUESTED** [review]
Just some minor comments

File: agent-ovs/lib/QosManager.cpp [hunk 1]
```diff
@@ -230,25 +240,157 @@ namespace opflexagent {
...
+        }
+        return boost::none;
+    }
+
+    uint8_t QosManager::resolveDscp(const optional<URI>& reqOpt) const {
+        URI req("");
+        if (reqOpt) {
+            req = reqOpt.get();
+        }
+        optional<shared_ptr<modelgbp::qos::Requirement>> qosReqOpt =
```
[the remark this answers is not in the document]
Comment (ansao-aci) [reply agent-ovs/lib/QosManager.cpp:292]: Done.

File: agent-ovs/lib/QosManager.cpp [hunk 2]
```diff
@@ -230,25 +240,157 @@ namespace opflexagent {
...
+            modelgbp::qos::Requirement::resolve(framework, req);
+
+        if (qosReqOpt){
+            shared_ptr<modelgbp::qos::Requirement> qosReq = qosReqOpt.get();
+
+            optional<shared_ptr<modelgbp::qos::RequirementToEgressRSrc> > rsEgress =
+                qosReq->resolveQosRequirementToEgressRSrc();
+            optional<URI> egressUri;
```
[the remark this answers is not in the document]
Comment (ansao-aci) [reply agent-ovs/lib/QosManager.cpp]: Done.

File: agent-ovs/lib/QosManager.cpp [hunk 3]
```diff
@@ -661,6 +832,9 @@ namespace opflexagent {
 
     void QosManager::updateEpgPolicyMap(const URI& epg, const URI& newReq){
         clearEpgEntry(epg);
+        LOG(INFO) << "epgToReq, epg: : " <<epg
```
[the remark this answers is not in the document]
Comment (ansao-aci) [reply agent-ovs/lib/QosManager.cpp]: Done.

**Review by tomflynn — APPROVED** [review]

**Pull request merged**